### PR TITLE
Expect autoformat delimiters to have a space character preceding them.

### LINKED ISSUE #2388: Markdown autoformat italics bug
Sorry if this was fixed in `latest` or not, but I wanted to make to discuss this.

From a user:

> I was writing some notes and tried writing wrote word1_word2_word3 and while writing it, it did word1word2word3 and word 2 was in italics. I am assuming that's because of the underlines but I am not sure this is intentional behavior. Thanks! 

STEPS TO REPRODUCE
1. Type: word_1 and word_2

EXPECTED RESULTS
* word_1 and word_2

ACTUAL RESULTS
* word*1 and word*2

Ref: https://github.com/mozilla/notes/issues/595

I wonder if auto-format should only format if there are spaces around `_` or if it starts the line?

### PR DESCRIPTION
**Pull request by maxbarnas**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Fix (autoformat): Formatting will not be applied to snake_case anymore. Closes #2388.

### CODE REVIEW

File: packages/ckeditor5-autoformat/src/autoformat.js [hunk 1]
```diff
@@ -80,17 +80,17 @@ export default class Autoformat extends Plugin {
 		if ( commands.get( 'bold' ) ) {
 			const boldCallback = getCallbackFunctionForInlineAutoformat( this.editor, 'bold' );
 
-			inlineAutoformatEditing( this.editor, this, /(\*\*)([^*]+)(\*\*)$/g, boldCallback );
-			inlineAutoformatEditing( this.editor, this, /(__)([^_]+)(__)$/g, boldCallback );
+			inlineAutoformatEditing( this.editor, this, /(?:^|\s)(\*\*)([^*]+)(\*\*)$/g, boldCallback );
```
Comment (maxbarnas) [inline packages/ckeditor5-autoformat/src/autoformat.js:83]: Could add a comment explaining why I added this here. Not sure if that's necessary.

File: packages/ckeditor5-autoformat/tests/autoformat.js [hunk 2]
```diff
@@ -577,24 +613,24 @@ describe( 'Autoformat', () => {
 
 			describe( 'should format', () => {
 				it( '* after', () => {
-					setData( model, '<paragraph><$text code="true">fo*o</$text>b*ar[]</paragraph>' );
+					setData( model, '<paragraph><$text code="true">fo*o</$text>b *ar[]</paragraph>' );
```
Comment (maxbarnas) [inline packages/ckeditor5-autoformat/tests/autoformat.js:616]: I've changed the test to let it format according to PR changes.

**Comment by Reinmar** [comment]
All looks fine, but it'd be good to have a clarity on cases that you ignored and that will become possible followups.

Some that I can think of:

*   The reversed case – typing `*` in `foo *^bar**`.
*   The non-white-space delimiters:
    *   typing `*` in `foo **bar*^.`.
    *   typing `*` in `foo (**bar*^`.
    *   typing `*` in `foo (*^bar**)`.

What else did you skip? It'd be good to have tickets for these scenarios.

**Comment by maxbarnas** [comment]
Follow-up tickets for future improvements: #8629 and #8630.

**Pull request merged**